### DISCUSSION TITLE
Zapier - Stripe / Fatture in Cloud

### DISCUSSION THREAD

**Discussion by Dreamerscommunity** [fattureincloud/api#446] [Q&A] [answered]
Ciao a tutti,

Complimenti per il rilascio del connettore Zapier, super utile. 

Sto cercando di connettere Stripe con Fatture in Cloud tramite Zapier, con l'obiettivo di inviare direttamente la fattura elettronica nel cassetto fiscale di un utente al momento della sottoscrizione di una membership sul nostro sito.

Chi acquista potrebbe essere sia un privato (quindi lo SDI di default dovrebbe essere 000000) oppure un'azienda (l'utente lo potrà scegliere su Stripe, al momento del pagamento, inserendo quindi il proprio SDI)

Ci sono alcuni campi che non saprei come compilare lato Zapier, ed in particolare

The Id of the document (required)
Value of TipoCassa Used (optional)
If set to true the invoice will not be sent to the SDI
Potresti aiutarmi a configurarli?

Grazie davvero e a presto

Marco

**Comment by valmoz** [comment]
Buongiorno @Dreamerscommunity 
siamo contenti che le piaccia il nostro connettore Zapier.

Qui di seguito le allego della documentazione che potrebbe esserle utile:

1. [Creazione di Issued Document](https://developers.fattureincloud.it/docs/guides/invoice-creation) (con esempi Zapier)
2. [Gestione delle fatture elettroniche](https://developers.fattureincloud.it/docs/guides/e-invoice-management)
3. [API Reference](https://developers.fattureincloud.it/api-reference) qui potrà trovare l'elenco dei campi accettati dalle nostre API e vedere degli esempi
4. [Zapier Quickstart](https://developers.fattureincloud.it/docs/quickstarts/zapier-quickstart)

Per poterla aiutare con i parametri in maniera puntuale, è necessario indicare per ciascuno:
- A quale azione Zapier si riferisce il parametro
- Qual è l'id del parametro (quello dentro le parentesi quadre, ad esempio "[data.entity.id]")
- Se possibile, il flusso che sta cercando di implementare (la sequenza delle action Zapier)

Per quanto riguarda i parametri che ci ha indicato, provo a fornirle delle indicazioni generiche in attesa di quanto chiesto qui sopra.
- L'ID del documento è l'identificativo univoco del documento all'interno di Fatture in Cloud; è obbligatorio solo in fase di lettura (quindi se sta utilizzando la _Get Issued Document_) ma deve essere omesso in fase di creazione del documento (in quanto viene assegnato in fase di creazione).
- Non saprei dirle a quale campo si riferisce, aspetto l'ID e l'azione coinvolti
- Questo penso che si riferisca al [Dry Run](https://developers.fattureincloud.it/docs/guides/e-invoice-management#pirate_flag-the-dry-run-flag) da utilizzare nel metodo _Send E-Invoice_, ma per esserne sicuro devo ottenere l'ID del parametro. Questo flag deve essere a "true" durante i test (per evitare di inviare una fattura di prova al Sistema di Interscambio), mentre deve essere omesso o impostato a "false" in produzione.

Restiamo a disposizione
Mauro

**Reply by Dreamerscommunity** [reply]
Buonasera @valmoz 

Grazie infinite per il supporto. Non sono super tecnico quindi cercherò di rispondere al meglio, aiutandomi con degli screenshot

Il flusso è il seguente: **Nuovo abbonamento sottoscritto su Stripe --> Emissione automatica Fattura elettronica tramite FIC nel cassetto fiscale del cliente (privato o azienda che sia)** 

Questa è la situazione attuale (Vedere screenshot)

![image](https://github.com/fattureincloud/api/assets/170263689/a325bd4f-6085-4940-87b0-4cc9d4de97bc)


- ID of the company l'ho omesso ora ma ho capito senza problemi quale selezionare
- Value of Tipo di cassa --> non saprei proprio cosa inserire
- Causale pagamento --> qui come si può vedere ho sovrascritto perchè vorrei il valore fosse sempre il medesimo
- If set to true the invoice will not be sent to the SDI -->  anche qui non saprei cosa scrivere

Grazie davvero

Marco

**Reply by valmoz** [reply]
Buongiorno @Dreamerscommunity 
Il flusso che ha implementato finora è errato.

Come descritto nelle guide che le ho passato, il metodo [Send E-Invoice](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/issued_documents/-document_id-/e_invoice/send) si occupa solamente di inviare all'SDI una fattura elettronica già esistente.
Il metodo per creare una fattura elettronica è invece il metodo [Create Issued Document](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/issued_documents), che le permetterà di inserire una nuova fattura in Fatture in Cloud.
Quindi il flusso completo sarebbe:

1. Trigger Stripe
2. Create Issued Document -> con questa creerà la fattura su Fatture in Cloud, ma non invierà all'SDI
3. Send E-Invoice -> con questa invierà la fattura creata in precedenza all'SDI

Poiché il metodo Create E-Invoice ha un body molto complesso, le suggerisco di seguire passo passo la guida che le ho allegato in precedenza (Creazione di Issued Document).

Grazie
Mauro

**Reply by valmoz** [reply]
Buongiorno @Dreamerscommunity
è riuscito a risolvere?

Grazie
Mauro

**Comment by Dreamerscommunity** [comment]
Buonasera @valmoz 

Mi scuso ma sto riprendendo ora in mano il tema; ho letto una parte del
manuale, ma non essendo molto tecnico sto facendo un po' fatica.

Lato zapier, sarebbe corretto questo procedimento?

[image](https://github.com/fattureincloud/api/assets/170263689/ff9681e6-cc63-42e0-8be1-954674bfac35)

Come le dicevo, lato Stripe ho ancora la vecchia ragione sociale (Che
corrisponde ad un'associazione). A breve, farò lo switch dell'account
sull'azienda, e quindi inizierò ad emettere le fatture.

Ho modo in qualche modo, di testare l'integrazione prima dell'effettivo
switch?

Grazie

Marco

Il giorno mar 28 mag 2024 alle ore 14:01 Mauro Valota <
***@***.***> ha scritto:

**Reply by Dreamerscommunity** [reply]
Buongiorno @valmoz 

Potrebbe darmi un riscontro? Grazie davvero

Marco

**Reply by valmoz** [reply]
Buongiorno @Dreamerscommunity 
sì, dovrebbe essere corretto. 

Per testare può procedere in due modi:
1. Utilizzare la sua company vera, facendo attenzione a inserire il [Dry Run Flag](https://developers.fattureincloud.it/docs/guides/e-invoice-management) in modo tale da non inviare la fattura al SDI ma solo simulare l'invio
2. Come il punto 1, ma rimuovendo lo step di invio SDI in modo tale da vedere solamente la fattura generata senza simulare l'invio
3. Utilizzare una company di test, con licenza Trial per motivi di testing, in modo tale da non sporcare la sua company reale. Nella company Trial l'invio all'SDI è disabilitato di default, ma le consiglierei comunque di mettere il Dry Run come per il caso 1 per sicurezza.

Nota Bene: qualora dovesse per errore inviare effettivamente la fattura di test all'SDI, la fattura risulterebbe non più cancellabile e sarebbe necessario emettere Nota di Credito per correggere l'invio errato della fattura.

Grazie
Mauro

**Reply by valmoz** [reply]
@Dreamerscommunity le consiglio comunque di controllare il metodo New Subscriptions di Stripe.. purtroppo non conosco le API di Stripe e quindi non posso dirle se è l'evento corretto, ma dal nome non mi sembra essere l'evento che le serve.

**Reply by Dreamerscommunity** [reply]
Bungiorno @valmoz grazie mille. Proverò appena pronto l'account stripe.

L'evento "New Subscription" credo sia quello corretto, in quanto è l'unica tipologia di prodotto che vendiamo su Stripe (abbonamenti).

**Comment by Dreamerscommunity** [comment]
Buonasera @valmoz 

Sto proseguendo nel setup e credo di essere a buon punto. Potrei chiederle gentilmente altre 2 info?

- All'interno del body della invoice, quali sono i campi obbligatori che devo per forza inviare tramite zapier?
- Come posso stabilire la numerazione della fattura, in automatico? Tipo vorrei che la prima fosse M1, la seconda M2 etc..

Grazie mille

Marco

**Reply by valmoz** [reply, marked as answer]
Buongiorno @Dreamerscommunity 
purtroppo non ci è possibile fornirle un elenco dettagliato dei campi obbligatori nel documento, in quanto dipende dalla tipologia di documento che sta cercando di creare.
Le posso però suggerire alcune strategie per completare questo step:

1. Seguire la nostra guida [Invoice Creation](https://developers.fattureincloud.it/docs/guides/invoice-creation) che elenca passo passo tutte le operazioni che è necessario fare per creare una fattura, includendo anche esempi Zapier
2. Creare una fattura manualmente su Fatture in Cloud così come la si desidera, e recuperare il body della fattura tramite una chiamata [List Issued Documents](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/issued_documents) o una [Get Issued Document](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/issued_documents/-document_id-); in tal caso le consigliamo di procedere con i tentativi tramite [API Reference](https://developers.fattureincloud.it/docs/api-reference) e [Autenticazione Manuale](https://developers.fattureincloud.it/docs/authentication/manual-authentication). Il JSON restituito sarà identico a quello necessario per creare la stessa fattura (al netto di campi che non possono essere conosciuti a priori, come l'ID della fattura), e basterà mappare i campi a quelli di Zapier.
3. Quando effettua una prova di creazione, la nostra API restituisce una risposta contenente la motivazione di un eventuale errore. Le suggeriamo quindi di effettuare delle prove (più comodo da API Reference, ma anche da Postman o Zapier) andando a risolvere i diversi errori di validazione che dovrebbero sorgere.

Per quanto riguarda la domanda sulla numerazione, come indicato nella nostra [guida](https://developers.fattureincloud.it/docs/guides/invoice-creation#two-step-two-the-invoice-data) è possibile omettere il campo "number", in tal caso Fatture in Cloud procederà con la creazione della fattura con numerazione progressiva rispetto a quanto già presente nel suo account.

Restiamo a disposizione per altri dubbi.

Grazie
Mauro

**Reply by valmoz** [reply]
Buongiorno @Dreamerscommunity 
abbiamo risposto alle sue domande? Possiamo chiudere la discussione?

Grazie
Mauro

**Reply by Dreamerscommunity** [reply]
Ciao @valmoz, spero tutto bene.
Ho quasi terminato il setup, ma purtroppo riscontro un errore che non riesco a risolvere 
![image](https://github.com/user-attachments/assets/7d275115-8cdc-4f44-b742-56301db1996a)

Mi dice "data.entity.country" field is not valid. Attualmente lo passiamo come "IT" (Come da tracciato del pagamento zapier).

Il supporto Zapier non riesce aiutarmi. Hai idea di come potrei risolvere?

Un'altra domanda: sapresti dirmi come potrei far in modo che Fatture in Cloud emetta una fattura con la numerazione che vogliamo noi? Grazie

Marco

**Reply by valmoz** [reply]
Per favore @Dreamerscommunity le chiederei di evitare di riaprire una discussione già chiusa, optando piuttosto per aprirne una nuova. 

Grazie

**Reply by Dreamerscommunity** [reply]
Procedo, grazie e scusa :)

**Reply by valmoz** [reply]
Nessun problema, aggiungo il link all'altra discussione per completezza: https://github.com/fattureincloud/api/discussions/466
